### ISSUE TITLE
noDescendingSpecificity broken on nested CSS [beta]

### ISSUE THREAD

**Issue by hyperknot** [biomejs/biome#6050]
### Environment information

```bash
Playground repros
```

### Rule name

lint/style/noDescendingSpecificity

### Playground link

https://next.biomejs.dev/playground/?code=LgBjAG8AbgB0AGEAaQBuAGUAcgAgAHsACgAgACAAdQBsACwACgAgACAAbwBsACAAewAKACAAIAAgACAAbQBhAHIAZwBpAG4ALQBiAG8AdAB0AG8AbQA6ACAAMQByAGUAbQA7AAoAIAAgAH0ACgAKACAAIAB1AGwALAAKACAAIABvAGwAIAB7ACAAIAAgACAACgAgACAAIAAgAHUAbAAsAAoAIAAgACAAIABvAGwAIAB7AAoAIAAgACAAIAAgACAAbQBhAHIAZwBpAG4ALQB0AG8AcAA6ACAAMAA7AAoAIAAgACAAIAB9AAoAIAAgAH0ACgB9AA%3D%3D&language=css&ruleDomains.solid=recommended

### Expected result

Code should be OK.

### Code of Conduct

- [x] I agree to follow Biome's Code of Conduct

**Comment by Th3S4mur41** [comment]
Facing the same issue with nested media query

```css
section {
  & img {
    ...
  }
  @media (width >= 1024px) {
    & img {
      ...
    }
  }
}
```

**Comment by JonnyWhitney** [comment]
Another small example of this breaking:

```
.component-a {
  & > .comopnent-a-sub {
    background-color: red;

    &:active {
      background-color: blue;
    }
  }
}

.component-b {
  background-color: green;

  &:active {
    background-color: yellow;
  }
}
```

It is comparing both `&:active` when `&` is referencing two different classes.

**Comment by ematipico** [comment]
Closing, it seems we fixed the bug
